src/Components/Endpoints/src/Builder/ResourcePreloadCollection.cs

```diff
@@ -0,0 +1,94 @@
+
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+internal class ResourcePreloadCollection
+{
+    private readonly Dictionary<string, StringValues> _storage = new();
+
+    public ResourcePreloadCollection(ResourceAssetCollection assets)
+    {
+        if (assets != null)
+        {
+            var headers = new List<(string? Group, int Order, string Value)>();
+            foreach (var asset in assets)
+            {
+                if (asset.Properties == null)
+                {
+                    continue;
+                }
+
+                // Use preloadgroup=webassembly to identify assets that should to be preloaded
+                string? group = null;
+                foreach (var property in asset.Properties)
+                {
+                    if (property.Name.Equals("preloadgroup", StringComparison.OrdinalIgnoreCase))
+                    {
+                        group = property.Value ?? string.Empty;
+                        break;
+                    }
+                }
+
+                if (group == null)
+                {
+                    continue;
+                }
+
+                var header = new StringBuilder();
+                header.Append('<');
+                header.Append(asset.Url);
+                header.Append('>');
+
+                int order = 0;
+                foreach (var property in asset.Properties)
+                {
+                    if (property.Name.Equals("preloadrel", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header.Append("; rel=").Append(property.Value);
+                    }
+                    else if (property.Name.Equals("preloadas", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header.Append("; as=").Append(property.Value);
+                    }
+                    else if (property.Name.Equals("preloadpriority", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header.Append("; fetchpriority=").Append(property.Value);
+                    }
+                    else if (property.Name.Equals("preloadcrossorigin", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header.Append("; crossorigin=").Append(property.Value);
+                    }
+                    else if (property.Name.Equals("integrity", StringComparison.OrdinalIgnoreCase))
+                    {
+                        header.Append("; integrity=\"").Append(property.Value).Append('"');
+                    }
+                    else if (property.Name.Equals("preloadorder", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (!int.TryParse(property.Value, out order))
+                        {
+                            order = 0;
+                        }
+                    }
+                }
```

Comment:
Could we move this into its own method? `TryCreateHeader` or something like that?